lib/champions_of_mirra_web/controllers/user_controller.ex

```diff
@@ -9,10 +9,9 @@ defmodule ChampionsOfMirraWeb.UserController do
 
   def get_opponents(conn, %{"device_client_id" => device_client_id}) do
     current_user = Accounts.get_user_by_device_client_id(device_client_id)
+    opponents = Accounts.get_all_usernames_and_ids() -- [%{id: current_user.id, username: current_user.username}]
 
-    json(conn, %{
-      users: Accounts.get_all_usernames_and_ids() -- [%{id: current_user.id, username: current_user.username}]
-    })
+    json(conn, Enum.map(opponents, &%{id: &1.id, username: &1.username}))
```

Comment:
There's no need to do that map function: it's exactly the same format as `opponents` already has (that's how `Accounts.get_all_usernames_and_ids()` returns it). What's accomplishing what you want is removing the wrapping `%{users: opponents}` map. So really the solution here is to just leave:
```
json(conn, opponents)
```